rules/no-disabled-tests.js

```diff
@@ -58,7 +85,19 @@ module.exports = {
             context.report({ message: 'Skipped test', node });
             break;
 
-          case 'pending':
+          case 'pending': {
+            const references = collectReferences(context.getScope());
+
+            if (
+              // `pending` was found as a local variable or function declaration.
+              references.locals.has('pending') ||
+              // `pending` was not found as an unresolved reference,
+              // meaning it is likely not an implicit global reference.
+              !references.unresolved.has('pending')
+            ) {
+              break;
```

Comment:
Let me know if this if statement is clear enough with the comment or if you see a clearer way to communicate this - I want this change to be as understandable and maintainable as possible. ❤️